test/arm/neon/reinterpret.c

```diff
@@ -0,0 +1,6760 @@
+ },
+    { {  INT8_C( 104), -INT8_C(  72),  INT8_C( 117), -INT8_C(  60),  INT8_C(  71),  INT8_C(  68),  INT8_C(  51),  INT8_C(  73) },
+ },
+    { {  INT8_C(  17),  INT8_C(  79), -INT8_C(  67),  INT8_C(  89), -INT8_C(  92), -INT8_C(   5), -INT8_C(  54),  INT8_C(  76) },
+ },
+    { { -INT8_C( 117), -INT8_C(  98), -INT8_C(  82),  INT8_C( 106), -INT8_C(  80),  INT8_C(  15), -INT8_C(  30),  INT8_C(  74) },
+ },
+    { { -INT8_C(  83),  INT8_C(  51), -INT8_C(   2),  INT8_C(  79),  INT8_C(  43), -INT8_C(  76),  INT8_C( 114), -INT8_C( 109) },
+ },
+    { {  INT8_C( 109), -INT8_C(  25),  INT8_C(  87), -INT8_C(  76),  INT8_C(  44), -INT8_C( 117), -INT8_C(   3),  INT8_C(  61) },
+ },
+    { { -INT8_C(  38), -INT8_C(  69), -INT8_C( 105),  INT8_C( 126), -INT8_C(  74),  INT8_C(  97), -INT8_C(  53),  INT8_C(  65) },
+ },
+    { { -INT8_C(   1),  INT8_C( 121), -INT8_C(  85), -INT8_C(  81), -INT8_C( 120), -INT8_C( 115), -INT8_C(   6),  INT8_C(  53) },
+ },
+
+  };
+
+  for (size_t i = 0 ; i < (sizeof(test_vec) / sizeof(test_vec[0])) ; i++) {
+    simde_int8x8_t a = simde_vld1_s8(test_vec[i].a);
+    simde_int8x8_private a_ = simde_int8x8_to_private(a);
+    simde_int64x1_t r = simde_vreinterpret_s64_s8(a);
+    simde_int64x1_private r_ = simde_int64x1_to_private(r);
+
+    simde_assert_equal_i(0, simde_memcmp(&r_, &a_, sizeof(r_)));
+  }
+
+  return 0;
+}
+
+static int
+test_simde_vreinterpret_u8_s8 (SIMDE_MUNIT_TEST_ARGS) {
+  struct {
+    int8_t a[8];
+  } test_vec[] = {
+    { {  INT8_C(  73), -INT8_C(  58), -INT8_C(  54),  INT8_C(   0),      INT8_MAX, -INT8_C(  14), -INT8_C(  96),  INT8_C( 124) },
+ },
+    { {  INT8_C(  61), -INT8_C(  81), -INT8_C(  61),  INT8_C(  43),  INT8_C(  92),  INT8_C(  16), -INT8_C(  94),  INT8_C( 125) },
+ },
+    { { -INT8_C(  46),  INT8_C(  72), -INT8_C(  42),  INT8_C(  56),  INT8_C(   3),  INT8_C(  67), -INT8_C(  58), -INT8_C(  39) },
+ },
+    { {  INT8_C(  44), -INT8_C(  82),  INT8_C(  29), -INT8_C(  51),  INT8_C(  70),  INT8_C(  74), -INT8_C(  65), -INT8_C( 113) },
+ },
+    { {  INT8_C(  17), -INT8_C( 119), -INT8_C( 112), -INT8_C( 112),  INT8_C( 124),  INT8_C(  48),  INT8_C(  12), -INT8_C(  71) },
+ },
+    { { -INT8_C(  33), -INT8_C(  49), -INT8_C(  28),  INT8_C(  59), -INT8_C(  33), -INT8_C( 122), -INT8_C(  71), -INT8_C(  78) },
+ },
+    { { -INT8_C(  50), -INT8_C( 113), -INT8_C(  22), -INT8_C(  47), -INT8_C(  46), -INT8_C(  80), -INT8_C(  86), -INT8_C(   1) },
+ },
+    { {  INT8_C(  94), -INT8_C(  57), -INT8_C(  52), -INT8_C(  92),  INT8_C(  18), -INT8_C( 116),  INT8_C(  52),  INT8_C(  35) },
+ },
+
+  };
+
+  for (size_t i = 0 ; i < (sizeof(test_vec) / sizeof(test_vec[0])) ; i++) {
+    simde_int8x8_t a = simde_vld1_s8(test_vec[i].a);
+    simde_int8x8_private a_ = simde_int8x8_to_private(a);
+    simde_uint8x8_t r = simde_vreinterpret_u8_s8(a);
+    simde_uint8x8_private r_ = simde_uint8x8_to_private(r);
+
+    simde_assert_equal_i(0, simde_memcmp(&r_, &a_, sizeof(r_)));
```

Comment:
Any idea why there are `unused_variable` warnings here? I'm not getting them on my end despite compiling with `-Wall`, so I'm sorta confused. Plus, I'm taking the address of it here, so it should be getting used...

Reply:
(The warnings are on `a_`)

Reply:
Wait, nevermind, I think I've solved it. I'll push it tomorrow with the native aliases.